byway/src/main/webapp/routepage.js

```diff
@@ -70,112 +75,117 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
     }
   });
 }
-
+ 
 /** Clear the stops panel in the html */
 function clearStops(){
   const stopList = document.getElementById('stop-list');
   if(stopList != null){
     stopList.innerText = "";
   }
 }
-
+ 
 /** Get the new list of stops from datastore onload */
 function getStopsOnload(){
   clearStops();
   fetch('/api/stop')
   .then(response => response.json())
   .then((stopsResponse) => {
-    stopsResponse.forEach((stop)=>{
-      stops.push(stop);
-    });
+    if(stopsResponse != null){
+      stopsResponse.forEach((stop)=>{
+        stops.add(stop);
+        waypoints.add({location:stop});
+      });
+    }
+    calcRoute();
     renderStopsList();
   });
 }
-
+ 
 /** Re-render stop list synchronously */
 function renderStopsList(){
   clearStops();
+  const stopList = document.getElementById('stop-list');
   stops.forEach((stop)=>{
-    renderStop(stop);
+    stopList.appendChild(createStopButton(stop));
   })
 }
-
+ 
 /** Render stop list  
  *  @param {String} stop a String to add as a button in the schedule panel in the html
+ *  @return {button} stopBtn a button showing a selected stop
  */
-function renderStop(stop){
-  const stopList = document.getElementById('stop-list');
-  let btn = document.createElement('button');
-  btn.innerText = stop;
-  btn.className =  "btn rec-btn";
-  btn.addEventListener("click", function() {
-    updateStops(stop, "remove");
+function createStopButton(stop){
+  const stopBtn = document.createElement('button');
+  stopBtn.innerText = stop;
+  stopBtn.className =  "btn rec-btn";
+  stopBtn.addEventListener("click", function() {
+    // remove place from stops and waypoints set
+    stops = new Set([...stops].filter(stopObj => stopObj!= stop));
```

Comment:
nit: spacing towards the end when using the not equal operator

Reply:
Fixed.